### ISSUE TITLE
Linking Devices While Mapping Is Dangerously Error-Prone

### ISSUE THREAD

**Issue by rosieposieeee** [space-wizards/space-station-14#28274]
## Description

While mapping, which is already a tedious process, device links tend to become timebombs if you aren't careful.

- If you delete ANYTHING that is linked without removing the link first, you can no longer save the map. If you don't remember what you deleted, you have to randomly delete buttons until you can save. You now have to relink all that shit because there is no undo.

- You will sometimes, for reasons I don't understand, get 'invalid' links in a map file that successfully saves. You then have to edit the yml manually and hope you don't cause even more problems. (guess what i did this month to a massive update!!!!!!!!!!!!!)

**Additional context**

@IamVelcroboy can probably answer questions about it better than I can.

**Comment by rosieposieeee** [comment]
![image](https://github.com/space-wizards/space-station-14/assets/52761126/9a2f7429-d1bf-4cef-9fae-6517f8324e71)

also worth noting that when you DO fuck up fixing this, the error it gives you is completely unusable. this only throws on shutdown of the map, and this is some kind of per-session uid, so i have no fucking idea which entity this actually is

**Comment by IamVelcroboy** [comment]
Yeah, this seriously make remodeling sections of a map a major pain, as if mapping needed more tedium. Anyone who can make deleted entities be auto-removed from any list they are linked to would be a hero.

**Comment by IProduceWidgets** [comment]
That was an accidental issue link, please ignore. (this is still an issue, and I think is important, it just wasn't the specific device linking issue I was trying to link to)

**Comment by IProduceWidgets** [comment]
I may have fixed this here: https://github.com/space-wizards/RobustToolbox/pull/5587 but Im not confident enough to close the issue yet
